### ISSUE TITLE
returnurl parameter on login page is ignored if already logged in

### ISSUE THREAD

**Issue by SimonAnnetts** [cloudscribe/cloudscribe#935]
If a link takes you to the login page, but you are already logged in then the returnurl parameter is not honoured.

Example:
Not logged in: Link takes you to login page. User logs in, then is redirected to returnurl.
Logged in:  Link takes you to login page, but you are immediately redirected to the home page.

**Comment by CrispinF** [comment]
@SimonAnnetts why would a link ever be to the login page with a returnurl parameter? Can you explain the use case here?
The original link should be to the destination page, then the user should be redirected to the login page if not already logged in, with the returnurl parameter.

**Comment by SimonAnnetts** [comment]
There are cases in a web application (we have a specific one I won't mention here) where if a user tries to access a protected page and if they are not logged in then they are redirected to a simple content page (rather than the login page) that explains a bit about the protected service they are trying to access and tells them they need to log in. This simple content page has a login button on it which points to 
/account/login?returnurl=%2Fmy-protected-area
so that the user can then login and be redirected back to the protected page.
This all works fine.
However if the user lands on the above simple content page via a bookmark say, and clicks login when they are already still logged in then they are dumped at the home page and the redirect part is ignored. This is not a good experience for the user!

**Comment by markidox** [comment]
Perhaps an alternative solution / user experience could be that the login button in question would only show should the user not be logged in? Then they couldn't click it if they're logged in.

**Comment by CrispinF** [comment]
You're quite right in principle @markidox but Simon's situation is as it is, and it's possible to improve this behaviour so I think we should do it.

**Comment by JimKerslake** [comment]
testable on cs.current sites now
